homeassistant/components/sensor/hue.py

```diff
@@ -0,0 +1,513 @@
+"""
+This component provides sensor support for the Philips Hue system.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.hue/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+
+from aiohue.sensors import (ZGP_SWITCH_BUTTON_1, ZGP_SWITCH_BUTTON_2, ZGP_SWITCH_BUTTON_3,
+                            ZGP_SWITCH_BUTTON_4, ZLL_SWITCH_BUTTON_1_LONG_RELEASED,
+                            ZLL_SWITCH_BUTTON_1_SHORT_RELEASED, ZLL_SWITCH_BUTTON_2_LONG_RELEASED,
+                            ZLL_SWITCH_BUTTON_2_SHORT_RELEASED, ZLL_SWITCH_BUTTON_3_LONG_RELEASED,
+                            ZLL_SWITCH_BUTTON_3_SHORT_RELEASED, ZLL_SWITCH_BUTTON_4_LONG_RELEASED,
+                            ZLL_SWITCH_BUTTON_4_SHORT_RELEASED, TYPE_CLIP_GENERICSTATUS,
+                            TYPE_CLIP_HUMIDITY, TYPE_CLIP_LIGHTLEVEL, TYPE_CLIP_TEMPERATURE,
+                            TYPE_ZGP_SWITCH, TYPE_ZLL_SWITCH, TYPE_ZLL_LIGHTLEVEL,
+                            TYPE_ZLL_TEMPERATURE, TYPE_CLIP_SWITCH)
+import homeassistant.components.hue as hue
+from homeassistant.components.hue.const import (ATTR_DARK, ATTR_DAYLIGHT, ATTR_LAST_UPDATED,
+                                                ICON_REMOTE, UOM_HUMIDITY, UOM_ILLUMINANCE)
+from homeassistant.components.sensor import (DEVICE_CLASS_HUMIDITY,
+                                             DEVICE_CLASS_ILLUMINANCE,
+                                             DEVICE_CLASS_TEMPERATURE)
+from homeassistant.const import (ATTR_BATTERY_LEVEL, TEMP_CELSIUS)
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['hue']
+SCAN_INTERVAL = timedelta(seconds=1)
+
+ALL_SENSORS = [TYPE_CLIP_GENERICSTATUS, TYPE_CLIP_HUMIDITY, TYPE_CLIP_LIGHTLEVEL,
+               TYPE_CLIP_SWITCH, TYPE_CLIP_TEMPERATURE, TYPE_CLIP_HUMIDITY, TYPE_ZGP_SWITCH,
+               TYPE_ZLL_LIGHTLEVEL, TYPE_ZLL_SWITCH, TYPE_ZLL_TEMPERATURE]
+HUE_SENSORS = [TYPE_ZGP_SWITCH, TYPE_ZLL_LIGHTLEVEL, TYPE_ZLL_SWITCH, TYPE_ZLL_TEMPERATURE]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Old way of setting up Hue.
+
+    Can only be called when a user accidentally mentions hue platform in their
+    config. But even in that case it would have been ignored.
+    """
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the Hue sensors from a config entry."""
+    bridge = hass.data[hue.DOMAIN][config_entry.data['host']]
+    cur_sensors = {}
+
+    # Hue updates all devices via a single API call.
```

Comment:
This looks like a straight copy from the hur light platform and it's also the same as for the binary sensor platform. We should break out a general function that can be imported and called for all platforms when setting up the platform entry.